SetUserDataProcessingStatusActivity/handler.ts

```diff
@@ -0,0 +1,170 @@
+  kind: t.literal("SUCCESS"),
+  value: UserDataProcessing
+});
+export type ActivityResultSuccess = t.TypeOf<typeof ActivityResultSuccess>;
+
+// Activity failed because of invalid input
+const ActivityResultInvalidInputFailure = t.interface({
+  kind: t.literal("INVALID_INPUT_FAILURE"),
+  reason: t.string
+});
+export type ActivityResultInvalidInputFailure = t.TypeOf<
+  typeof ActivityResultInvalidInputFailure
+>;
+
+// Activity failed because of an error on a query
+const ActivityResultQueryFailure = t.intersection([
+  t.interface({
+    kind: t.literal("QUERY_FAILURE"),
+    reason: t.string
+  }),
+  t.partial({ query: t.string })
+]);
+export type ActivityResultQueryFailure = t.TypeOf<
+  typeof ActivityResultQueryFailure
+>;
+
+export const ActivityResultFailure = t.taggedUnion("kind", [
+  ActivityResultQueryFailure,
+  ActivityResultInvalidInputFailure
+]);
+export type ActivityResultFailure = t.TypeOf<typeof ActivityResultFailure>;
+
+export const ActivityResult = t.taggedUnion("kind", [
+  ActivityResultSuccess,
+  ActivityResultFailure
+]);
+
+export type ActivityResult = t.TypeOf<typeof ActivityResult>;
+
+const logPrefix = `SetUserDataProcessingStatusActivity`;
+
+function assertNever(_: never): void {
+  throw new Error("should not have executed this");
+}
+
+/**
+ * Logs depending on failure type
+ * @param context the Azure functions context
+ * @param failure the failure to log
+ */
+const logFailure = (context: Context) => (
+  failure: ActivityResultFailure
+): void => {
+  switch (failure.kind) {
+    case "INVALID_INPUT_FAILURE":
+      context.log.error(
+        `${logPrefix}|Error decoding input|ERROR=${failure.reason}`
+      );
+      break;
+    case "QUERY_FAILURE":
+      context.log.error(
+        `${logPrefix}|Error ${failure.query} query error |ERROR=${failure.reason}`
+      );
+      break;
+    default:
+      assertNever(failure);
+  }
+};
+
+export const createSetUserDataProcessingStatusActivityHandler = (
+  userDataProcessingModel: UserDataProcessingModel
+) => (context: Context, input: unknown) => {
+  /**
+   * Updates a UserDataProcessing record by creating a new version of it with a chenged status
+   * @param param0.currentRecord the record to be modified
+   * @param param0.nextStatus: the status to assign the record to
+   *
+   * @returns either an Error or the new created record
+   */
+  const saveNewStatusOnDb = ({
+    currentRecord,
+    nextStatus
+  }: {
+    currentRecord: UserDataProcessing;
+    nextStatus: UserDataProcessingStatus;
+  }): TaskEither<ActivityResultQueryFailure, UserDataProcessing> =>
+    tryCatch(
+      () =>
+        userDataProcessingModel.createOrUpdateByNewOne({
+          ...currentRecord,
+          status: nextStatus
+        }),
+      (err: Error) => {
+        return ActivityResultQueryFailure.encode({
+          kind: "QUERY_FAILURE",
+          query: "userDataProcessingModel.createOrUpdateByNewOne",
+          reason: err.message
+        });
+      }
```

Comment:
Can we use `fromQueryEither` in this case?

Reply:
I agree, but still `fromQueryEither` isn't as generic as to be placed in a dedicated module. I'll just copy it over though, because I think it's more eloquent